com.unity.multiplayer.mlapi/Runtime/Core/RpcBatcher.cs

```diff
@@ -10,29 +10,30 @@
 
 namespace MLAPI
 {
-    class MessageBatcher
+    internal class RpcBatcher
```

Comment:
I'm confused by this rename. MessageBatcher does not just batch RPCs currently. It also batches other messages like the send message events which get called in the NetworkedObject.Spawn function. Is this an oversight? MessageBatcher makes way more sense to me naming wise.

Reply:
does it? it looks very specific to RPCs (or RPC-like messages).
we had a discussion with Jeff and agreed that we could create other batcher classes and potentially create a more common batcher base class in the future and change type hierarhy (or make a composition out of it).
I think rename makes sense... 🥺

Reply:
RPC-like messages🙃? 
Just trying to point out how this looks like to someone who hasn't put much thought into this:
- Name suggests RPCs only
- Implementation is RPCs + a random? selection of events (why is time sync not queued for instance)

I guess to make this clean we would need a proper way to address messages which get queued. Then we can use that naming across the board.

Reply:
`MessageBatcher` is not correct either. Current implementation is not isolated and doesn't just do `byte[]` based packet/message batching — read the source code 👀

Reply:
As far as I know, the class considered for rename can only possibly handle MLAPI_SERVER_RPC and MLAPI_CLIENT_RPC. I would need modification to handle more.

It also seems to only be called from `void RpcQueueProcessor::RPCQueueSendAndFlush()`. So, the current usage of it is consistent with it being for RPCs only.

Whether this is the desired architecture for the long run, though, I don't know.

Reply:
'SpawnManager.SendSpawnCallForObject' uses this as well for something which is whether a client RPC nor a server RPC.

Reply:
I think we should keep `RpcBatcher` name change as is simply because it's more indicative atm. Having said that, we should revisit batcher implementation in the future and possibly generalize with a generic base class (or utilities) in the future.

Reply:
> 'SpawnManager.SendSpawnCallForObject' uses this as well for something which is whether a client RPC nor a server RPC.

I'll investigate this. This is a worry. The test is still good to go, as it's a step forward, but I want to understand the SpawnManager use. Will kick off offline discussions :-). Much thanks for pointing it out.